### PR TITLE
Remove callback for balance retrieval

### PR DESCRIPTION
**Pull request by flomnes**
Small refactor

### CODE REVIEW

File: src/solver/simulation/economy.cpp [hunk 1]
```diff
@@ -238,20 +238,20 @@ void Economy::incrementProgression(Progression::Task& progression)
...
-        ComputeFlowQuad(study, *pProblemesHebdo[0], callbackBalance, pNbWeeks);
+        // Retrieve weighted average balance for each area
+        const uint nbAreas = study.areas.size();
+        std::vector<AvgExchangeResults*> balance(nbAreas, nullptr);
+        for (uint areaIndex = 0; areaIndex < nbAreas; ++areaIndex)
+        {
+            const Data::Area* area = study.areas.byIndex[areaIndex];
+            variables.retrieveResultsForArea<Variable::Economy::VCardBalance>(&balance[areaIndex],
+                                                                              area);
+        }
```
Comment (guilpier-code) [inline src/solver/simulation/economy.cpp]: We could put line 246 to 253 into a function, named **Economy::retrieveBalance** and then have : 
```c++
void Economy::simulationEnd()
{
    if (!preproOnly && study.runtime->interconnectionsCount > 0)
    {
        std::vector<AvgExchangeResults*> balance = retrieveBalance(study);
        ComputeFlowQuad(study, *pProblemesHebdo[0], balance, pNbWeeks);
    }
}
```

File: src/solver/simulation/adequacy.cpp [hunk 2]
```diff
@@ -364,20 +364,20 @@ void Adequacy::incrementProgression(Progression::Task& progression)
...
-        ComputeFlowQuad(study, *pProblemesHebdo[0], callbackBalance, pNbWeeks);
+        // Retrieve weighted average balance for each area
+        const uint nbAreas = study.areas.size();
+        std::vector<AvgExchangeResults*> balance(nbAreas, nullptr);
+        for (uint areaIndex = 0; areaIndex < nbAreas; ++areaIndex)
+        {
+            const Data::Area* area = study.areas.byIndex[areaIndex];
+            variables.retrieveResultsForArea<Variable::Economy::VCardBalance>(&balance[areaIndex],
+                                                                              area);
+        }
```
Comment (guilpier-code) [inline src/solver/simulation/adequacy.cpp]: Save remark as for economy

File: src/solver/simulation/common-eco-adq.cpp [hunk 3]
```diff
@@ -58,12 +58,11 @@ static void RecalculDesEchangesMoyens(Data::Study& study,
...
-            if (balance)
+            assert(balance[j] && "Impossible to find the variable");
+            if (balance[j])
             {
                 problem.SoldeMoyenHoraire[i]->SoldeMoyenDuPays[j]
-                  = balance->avgdata.hourly[decalPasDeTemps];
+                  = balance[j]->avgdata.hourly[decalPasDeTemps];
             }
             else
             {
```
Comment (guilpier-code) [inline src/solver/simulation/common-eco-adq.cpp:68]: About the call to **retrieveAverageNTC** : 
```c++
int ret = retrieveAverageNTC(study, link, avgDirect, avgIndirect);
```
we could 2 calls to a single function : 
```c++
int ret = retrieveAverageNTC(study, link->directCapacities, avgDirect);
ret = retrieveAverageNTC(study, link->indirectCapacities, avgIndirect) && ret;
```
as the treatment is very similar for both direct and indirect capacities.
Or we can consider it is going too far in code factorization.

**Review by guilpier-code — CHANGES_REQUESTED** [review]

File: src/solver/simulation/common-eco-adq.cpp (hunk 3 above)
Comment (flomnes) [reply src/solver/simulation/common-eco-adq.cpp:68]: That looks good, I will do that

File: src/solver/simulation/adequacy.cpp (hunk 2 above)
Comment (flomnes) [reply src/solver/simulation/adequacy.cpp]: done

File: src/solver/simulation/economy.cpp (hunk 1 above)
Comment (flomnes) [reply src/solver/simulation/economy.cpp]: done

**Review by flomnes — COMMENTED** [review]
Accepted all remarks and applied the corresponding changes.

**Review by guilpier-code — APPROVED** [review]

**Pull request merged**